plugins/hg_hash_in_commit/__init__.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+## Copyright 2022 David Miguel Susano Pinto
+##
+## Permission is hereby granted, free of charge, to any person obtaining
+## a copy of this software and associated documentation files (the
+## "Software"), to deal in the Software without restriction, including
+## without limitation the rights to use, copy, modify, merge, publish,
+## distribute, sublicense, and/or sell copies of the Software, and to
+## permit persons to whom the Software is furnished to do so, subject to
+## the following conditions:
+##
+## The above copyright notice and this permission notice shall be
+## included in all copies or substantial portions of the Software.
+##
+## THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+## EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+## MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+## NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+## LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+## OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+## WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+class HgHashInCommitFilter:
+    def __init__(self, prefix=b"hg:"):
+        self._prefix = prefix
+
+    def commit_message_filter(self, commit_data):
+        assert "hg_hash" in commit_data and "desc" in commit_data
+
+        new_paragraph = b"\n\n"
+        commit_data['desc'] += (new_paragraph
+                                + self._prefix
+                                + commit_data["hg_hash"])
+
+
+def build_filter(opts):
+    # FIXME: having an empty prefix is not supported.
```

Comment:
Empty prefixes should be supportable using the [outline](https://github.com/frej/fast-export/issues/290#issuecomment-1172457687l) in #290 which should resolve this FIXME.